cmd/server/assets/login/registerphone.html

```diff
@@ -0,0 +1,89 @@
+{{define "login/register"}}
+<!doctype html>
+<html lang="en">
+
+<head>
+  {{template "head" .}}
+
+  {{template "firebase" .}}
+</head>
+
+<body>
+  {{template "navbar" .}}
+
+  <main role="main" class="container">
+    {{template "flash" .}}
+
+    <h1>SMS auth registration</h1>
+    <p id="subtitle">Use this page to register a phone number for 2-factor authentication.</p>
+
+    <div id="recaptcha-container" class="center-block"></div>
+
+    <div class="card mb-3">
+      <div class="card-body">
+        <div class="row form-group">
+          <label for="phone" class="col-sm-3">Phone number</label>
+          <div class="col-sm-9">
+            <div class="input-group p-0 shadow-sm">
+              <input type="tel" id="phone" name="phone" class="form-control" />
+            </div>
+            <small class="form-text text-muted">
+              Format: +1 123-456-7890. Standard SMS rates may apply.
+            </small>
+          </div>
+        </div>
+
+        <div class="row form-group">
+          <div class="offset-sm-3 col-sm-9">
+            <button id="register" class="btn btn-primary btn-block">Register</button>
+          </div>
+        </div>
+      </div>
+    </div>
+  </main>
+
+  {{template "scripts" .}}
+
+  <script>
+    window.recaptchaVerifier = new firebase.auth.RecaptchaVerifier(
+      'recaptcha-container',
+      { 'size': 'invisible' }
+    );
+
+    recaptchaVerifier.render()
+      .then(function (widgetId) {
+        window.recaptchaWidgetId = widgetId;
+      });
+
+    $(function () {
+      let $phone = $('#phone')
+
+      $('#register').on('click', function (event) {
+        firebase.auth().currentUser.multiFactor.getSession().then(function (multiFactorSession) {
+          // Specify the phone number and pass the MFA session.
+          var phoneInfoOptions = {
+            phoneNumber: $phone.val(),
+            session: multiFactorSession
+          };
+          var phoneAuthProvider = new firebase.auth.PhoneAuthProvider();
+          // Send SMS verification code.
+          return phoneAuthProvider.verifyPhoneNumber(
+            phoneInfoOptions, window.recaptchaVerifier);
+        }).then(
+          function (verificationId) {
+            var verificationCode = window.prompt('Please enter the verification ' +
+              'code that was sent to your mobile device.');
+            // TODO: enroll the credential
+          },
+          function (err) {
+            $('main').prepend(
+              '<div class="alert alert-warning" role="alert">'
+              + err.message + '</div>');
```

Comment:
This is subject to XSS (granted, the source is somewhat trusted). Building a jquery object and setting it's `text()` property is safer (see what I did in the other PR). This can be a follow up.

Reply:
I'm going to combine these into a function in .scripts per your other comment